src/Components/DotCMS/NewsWidgets.js

```diff
@@ -24,10 +24,13 @@ class NewsWidget extends React.Component {
         };
     }
 
-    getNewsItems(pagination) {
+    async getNewsItems(pagination) {
+        const langId = await DotCMSApi.languages.getId(
+            DotCMSApi.languages.getCode(window.location.search)
+        );
         const defaultEsParams = {
             itemsPerPage: this.props.itemsPerPage,
-            languageId: this.props.languageId,
+            languageId: langId || this.props.languageId,
```

Comment:
Why do we need the `languageId` prop in the state if we always will have that in the url?

Reply:
I think in `Dotcms` app in Edit mode, that is being passed through `props`